tests/components/ring/test_binary_sensor.py

```diff
@@ -1,24 +1,182 @@
 """The tests for the Ring binary sensor platform."""
 
+import time
+from unittest.mock import patch
+
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.ring.binary_sensor import RingEvent
+from homeassistant.components.ring.const import DOMAIN
+from homeassistant.components.ring.coordinator import RingEventListener
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er, issue_registry as ir
+from homeassistant.setup import async_setup_component
+
+from .common import setup_automation, setup_platform
+from .device_mocks import FRONT_DOOR_DEVICE_ID, INGRESS_DEVICE_ID
+
+from tests.common import async_fire_time_changed
+
+
+@pytest.mark.parametrize(
+    ("device_id", "device_name", "alert_kind", "device_class"),
+    [
+        pytest.param(
+            FRONT_DOOR_DEVICE_ID,
+            "front_door",
+            "motion",
+            "motion",
+            id="front_door_motion",
+        ),
+        pytest.param(
+            FRONT_DOOR_DEVICE_ID,
+            "front_door",
+            "ding",
+            "occupancy",
+            id="front_door_ding",
+        ),
+        pytest.param(
+            INGRESS_DEVICE_ID, "ingress", "ding", "occupancy", id="ingress_ding"
+        ),
+    ],
+)
+async def test_binary_sensor_without_deprecation(
+    hass: HomeAssistant,
+    mock_ring_client,
+    mock_ring_event_listener_class: RingEventListener,
+    freezer: FrozenDateTimeFactory,
+    device_id,
+    device_name,
+    alert_kind,
+    device_class,
+) -> None:
+    """Test the Ring binary sensors as if they were not deprecated."""
+    with patch(
+        "homeassistant.components.ring.binary_sensor.async_check_create_deprecated",
+        return_value=True,
+    ):
```

Comment:
Please avoid patching the integration itself in tests

Reply:
Done